Task Manager Repeat NOAA Crawler Error

Description:
Trying to run `DataFromNoaa` from Task Manager, with loop set to be true and interval 21600 seconds. got this error at initialization. 

Environment: 
cloudberry05 server
CentOS 7

Error msg:
```[08/06/2019-21:52:57] [INFO] [MainThread] [task_manager] [<module>]: Task manager is running now!
[08/06/2019-21:53:34] [INFO] [MainThread] [task_manager] [run]: A new task will be running!
[08/06/2019-21:53:34] [INFO] [DataFromNoaa1] [task_manager] [_thread_runner_]: TASK DataFromNoaa1 START!
[08/06/2019-21:53:34] [INFO] [DataFromNoaa1] [connection] [get_connection_status]: [DATABASE] HOST = cloudberry05.ics.uci.edu, CONNECTION COUNT = 8, MAXIMUM = 100
[08/06/2019-21:53:34] [INFO] [DataFromNoaa1] [data_from_noaa] [run]: start crawling
[08/06/2019-21:53:35] [INFO] [DataFromNoaa1] [noaa_crawler] [crawl]: saved file: 2019080700
[08/06/2019-21:53:37] [ERROR] [DataFromNoaa1] [task_manager] [_thread_runner_]: error: Traceback (most recent call last):
  File "backend/task_manager.py", line 197, in _thread_runner_
    target_func(*args)
  File "/extra/yicongh10/wildfires/backend/task/data_from_noaa.py", line 49, in run
    ugnd = extractor.extract(GRIBEnum.NOAA_WIND_U)
  File "/extra/yicongh10/wildfires/backend/data_preparation/extractor/grib_extractor.py", line 43, in extract
    prop_msg, = self.file_handler.select(**GRIBExtractor.NAMES[mode])
  File "pygrib.pyx", line 610, in pygrib.open.select
ValueError: no matches found

[08/06/2019-21:53:37] [INFO] [DataFromNoaa1] [task_manager] [_thread_runner_]: TASK DataFromNoaa1 END!
[08/06/2019-21:53:37] [ERROR] [DataFromNoaa1] [task_manager] [_thread_runner_]: error: Traceback (most recent call last):
  File "backend/task_manager.py", line 204, in _thread_runner_
    if not cls.running_threads[index_].loop:
IndexError: list index out of range
```

--- a/backend/task_manager.py
+++ b/backend/task_manager.py
@@ -157,8 +157,6 @@ class TaskManager:
         """Stops a looping function that was started with ThM.run(...)"""
         for i, thlis in enumerate(cls.running_threads):
             if thlis.th_name == thread_name:
-                cls.running_threads[i].loop = False
-                cls.running_threads.remove(thlis)
                 exc = ctypes.py_object(SystemExit)
                 res = ctypes.pythonapi.PyThreadState_SetAsyncExc(
                     ctypes.c_long(cls.running_threads[i].th.ident), exc)
@@ -167,7 +165,8 @@ class TaskManager:
                 elif res > 1:
                     ctypes.pythonapi.PyThreadState_SetAsyncExc(cls.running_threads[i].th.ident, None)
                     raise SystemError("PyThreadState_SetAsyncExc failed")
-                logger.info('TASK ' + cls.running_threads[i].th_name + ' KILLED!')
+                logger.info('TASK ' + str(cls.running_threads[i].th_name) + ' KILLED!')
+                cls.running_threads.remove(cls.running_threads[i])
                 break
 
     @classmethod
@@ -185,9 +184,9 @@ class TaskManager:
         Note: threading.Event() has not been taken into consideration and neither the
         other thread managing objects (semaphores, locks, etc.)"""
         try:
-            index_ = 0
-            if th_name in [thread_.th_name for thread_ in cls.running_threads]:
-                index_ += 1
+            # index_ = 0
+            # if th_name in [thread_.th_name for thread_ in cls.running_threads]:
+            #     index_ += 1
 
             while True:
                 logger.info('TASK ' + th_name + ' START!')
@@ -201,10 +200,19 @@ class TaskManager:
                 # 'END' means returned or crashed
                 logger.info('TASK ' + th_name + ' END!')
                 # whether a loop work?
-                if not cls.running_threads[index_].loop:
+
+                index_ = None
+                for i, thread_ in enumerate(cls.running_threads):
+                    if cls.running_threads[i].th_name == th_name:
+                        index_ = i
+
+                if index_ is not None and index_ < len(cls.running_threads):
+                    if not cls.running_threads[index_].loop:
+                        break
+                    if interval != 0:
+                        time.sleep(interval)
+                else:
                     break
-                if interval != 0:
-                    time.sleep(interval)
 
         except:
             logger.error("error: " + traceback.format_exc())
@@ -301,8 +309,6 @@ class TaskManager:
             running_thread = self.running_threads[int(stop_task_prompt)]
             self.stop_thread(running_thread.th_name)
             print("Task " + str(running_thread.th_name) + " has been stopped!\n")
-        except SystemExit:
-            pass
         except:
             print("Skipped, no task been terminated\n ")
 

